packages/components/pie-divider/src/index.ts

```diff
@@ -24,20 +26,37 @@ export class PieDivider extends LitElement implements DividerProps {
     @validPropertyValues(componentSelector, orientations, defaultProps.orientation)
     public orientation = defaultProps.orientation;
 
+    @property({ type: String })
+    public label = defaultProps.label;
+
     render () {
-        const { variant, orientation } = this;
+        const { variant, orientation, label } = this;
 
         const classes = {
-            'c-divider': true,
             'c-divider--inverse': variant === 'inverse',
             'c-divider--vertical': orientation === 'vertical',
+            'c-divider--labelled': label.length > 0 && orientation === 'horizontal',
```

Comment:
Maybe this could be pulled out into something like `const showLabel = label.length > 0 && orientation === 'horizontal';` which can then also be used in the html below

Reply:
agreed - lets try to keep the logic inside `classmap` super minimal :)